broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -332,11 +332,17 @@ public void onDiskSpaceAvailable() {
     final CompletableActorFuture<Void> completed = new CompletableActorFuture<>();
     actor.call(
         () -> {
-          context.setProcessingPaused(true);
-          if (context.getStreamProcessor() != null) {
-            context.getStreamProcessor().pauseProcessing().onComplete(completed);
-          } else {
-            completed.complete(null);
+          try {
+            context.setProcessingPaused(true);
+
+            if (context.getStreamProcessor() != null && !context.shouldProcess()) {
+              context.getStreamProcessor().pauseProcessing().onComplete(completed);
+            } else {
+              completed.complete(null);
+            }
+          } catch (final IOException e) {
```

Comment:
ioe?

Reply:
What happens then ? Do we return an error?

Reply:
The returned future is completed exceptionally. But the api to pause processing is async, so it is not returned to the client.

Reply:
async means you could complete it right with the error right? But might make it to complex idk

Reply:
Yes. It is completed with error. But the actuators that call this method does not wait for the future. It is just trigger and go.